### DISCUSSION TITLE
pyproject.toml dynamic [optional-]dependencies using "file = ..." - BETA status?

### DISCUSSION THREAD

**Discussion by bunny-therapist** [pypa/setuptools#3758] [Q&A]
I am currently using "file = requirements.txt" in dependencies and optional-dependencies in pyproject.toml and I was considering using it in a lot more projects in our group, since we have projects which currently open requirements.txt files and reads them, programmatically, inside of setup.py, and I wanted to replace this with just linking requirements.txt in pyproject.toml. Howeer, I have put that change on hold because of the beta warning; I don't want to risk making all our projects depend on something that later gets removed.

How stable is this feature? Can I expect it to remain in the future?

To be specific, I am talking about stuff like this in pyproject.toml:
```
[project]
...
dynamic = ["dependencies", "optional-dependencies"]

[tool.setuptools.dynamic.dependencies]
file = ["requirements.txt"]

[tool.setuptools.dynamic.optional-dependencies.tensorflow]
file = ["tensorflow-requirements.txt"]
```

**Comment by webknjaz** [comment]
“Beta” usually means that it expects end-user feedback. If there's no serious UX problems, it'll remain. At some point, it'll become stable as is, if there's no compelling reason to change it. I think, at this point, the adoption may still be low which is why it's still in beta. Maybe @abravalheri knows of a deadline for deciding that it's mature enough. FWIW, you can always pin the `setuptools` version in `build-system` to improve stability and reproducibility.

**Comment by bunny-therapist** [comment]
I have pinned setuptools in the project where I used this feature. I just do not want to use it in a large number of internal projects because doing that (and pinning setuptools) would mean that it would be harder to update setuptools versions later. So I dare to use it in singular projects (with pinned setuptools), I just do not dare to dare to introduce it in a lot of projects.
